include/os/freebsd/spl/sys/taskq.h

```diff
@@ -48,16 +48,16 @@ typedef uintptr_t taskqid_t;
 typedef void (task_func_t)(void *);
 
 typedef struct taskq_ent {
-	struct task	 tqent_task;
-	struct timeout_task tqent_timeout_task;
+	union {
+		struct task	 tqent_task;
+		struct timeout_task tqent_timeout_task;
+	};
 	task_func_t	*tqent_func;
 	void		*tqent_arg;
-	taskqid_t tqent_id;
+	taskqid_t	 tqent_id;
 	CK_LIST_ENTRY(taskq_ent) tqent_hash;
```

Comment:
there is no point using CK_LIST here. Employing CK_* routines is highly misleading as there is only locked usage. This can be switched to mere SLIST variants and shave another 8 bytes.

Reply:
SLIST won't be good since we need arbitrary removal.  Could be just a LIST though, but I suppose without any real differences.

Reply:
You de facto pay for the lookup on removal, but since this is a hash, all chains are expected to be super short (2-3 elements tops). If they are long, taskq_lookup is already slow and that needs fixing.